frontend/src/scenes/settings/project/SessionRecordingSettings.tsx

```diff
@@ -166,12 +172,31 @@ function NetworkCaptureSettings(): JSX.Element {
                     <LemonSwitch
                         data-attr="opt-in-capture-network-body-switch"
                         onChange={(checked) => {
-                            updateCurrentTeam({
-                                session_recording_network_payload_capture_config: {
-                                    ...currentTeam?.session_recording_network_payload_capture_config,
-                                    recordBody: checked,
-                                },
-                            })
+                            if (checked) {
+                                LemonDialog.open({
+                                    title: 'Enabling network payload capture',
+                                    description: <PayloadWarning />,
```

Comment:
Don't know an appropriate size but the gif makes it look like this could benefit from constricting the width of the modal:
```suggestion
                                    description: <PayloadWarning />,
                                    width: ??
```